### ISSUE TITLE
[HOLD for payment 2023-08-08] [$2000] IOS - Chat - Main composer is displayed when editing second message

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#17531]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Issue found when executing PR** https://github.com/Expensify/App/pull/17221

## Action Performed:
1. Open NewDot app
2. Open any chat
3. Send 2 messages
4. Start edit for both messages

## Expected Result:
Main compose box should not be displayed

## Actual Result:
Main compose box appears on the screen once I start to edit second message

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.0.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/232604613-64146c4c-9b6c-47c3-bbe3-839122ff6006.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0193911d4ae1a59365</li>
        <li>Upwork Job ID: 1648310272611299328</li>
        2023-07-17<li>Automatic offers: </li>
<ul>
<li> |  | 0</li>
<li>s-alves10 | Contributor | 25753192</li>
</ul></li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @zanyrenney (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by zanyrenney** [comment]
I can't reproduce this in production -- 
<img width="432" alt="2023-04-18_13-57-32" src="https://user-images.githubusercontent.com/56830058/232784547-66c894a0-7aca-4d2e-a789-2b3d11a28bbc.png">

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~0193911d4ae1a59365

**Comment by MelvinBot** [comment]
Current assignee @zanyrenney is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @aimane-chnaif (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @techievivek (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by prempratapsingh** [comment]
Hi, this is Prem from Upwork.

This issue appears to be related to some iOS front end level of failing check which is making the compose new message view active when user has the edit messaging view open and then switches back and forth between the messages list and message details view.

This requires analyzing the message details view workflow (viewDidAppear, disLayoutSubViews, etc) and related checks to see if some check is failing. The fix shouldn’t be much complicated, I believe.

I can start investigating this issue, once the Upwork contract is assigned. Let me know please.

Best,
Prem

**Comment by MelvinBot** [comment]
📣 @prempratapsingh! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/231263719-a8a3b7d3-36a0-4d7f-8ad4-cebe1f560fa4.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by chriscoderdr** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Chat - Main composer is displayed when editing second message
### What is the root cause of that problem?

toggleReportActionComposeView (which hides or shows the composer view) is being call with true when the first comment lost focus because the keyboardDidHideListener

### What changes do you think we should make in order to solve the problem?

As we are calling toggleReportActionComposeView when the user click save changes and cancel we don't need this call on keyboardDidHide.

But fixing that issue there is another related problem that we see:
If you're editing 2 comments or more at the same time and press save changes or cancel in one of them composer view will be displayed.

To fix that second problem, I propose we take away the responsibility to call toggleReportActionComposeView from each individual message edit (ReportActionItemMessageEdit) and instead take that to ReportActionsList so we can know if more than one edit is being done and call it accordingly.

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by MelvinBot** [comment]
📣 @chriscoderdr! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/231263719-a8a3b7d3-36a0-4d7f-8ad4-cebe1f560fa4.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by chriscoderdr** [comment]
Contributor details
Your Expensify account email: christiangomezdev@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01cd7e16173e014f7b

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by techievivek** [comment]
@aimane-chnaif Can you please have a look at the above proposal by @chriscoderdr

**Comment by zanyrenney** [comment]
Heading OOO, reapplying the Bug0 label for active management.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @sonialiap (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by techievivek** [comment]
Not overdue, waiting for @aimane-chnaif to have a look at the proposal.

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $2000

**Comment by techievivek** [comment]
Not overdue, we are waiting for the proposal.

**Comment by techievivek** [comment]
@aimane-chnaif Can you please have a look at the above proposal before we pass this issue to the Callstack team.

**Comment by techievivek** [comment]
Not overdue, @aimane-chnaif  will have a look at the proposal by today.

**Comment by techievivek** [comment]
@aimane-chnaif, did you have a look at the [proposal](https://github.com/Expensify/App/issues/17531#issuecomment-1538067818) by any chance?

**Comment by techievivek** [comment]
@aimane-chnaif I have removed the internal label for now because we have a proposal review in the process.

**Comment by techievivek** [comment]
@HezekielT Thanks for the video, I will wait for @aimane-chnaif to have a look at the updated proposal.

**Comment by techievivek** [comment]
@aimane-chnaif Gentle bump to take a look at the above branch and review the proposal? Thanks

**Comment by sonialiap** [comment]
@aimane-chnaif how are we doing, can we move forward with a proposal?

**Comment by sonialiap** [comment]
@aimane-chnaif bump

**Comment by sonialiap** [comment]
@aimane-chnaif we have a few proposals here, what do you think of them?

**Comment by s-alves10** [comment]
@aimane-chnaif 

I merged the branch with the latest main. https://github.com/s-alves10/App/tree/fix/issue-17531
Thank you

**Comment by techievivek** [comment]
Bumping @aimane-chnaif to take a look at the updated branch to test the proposal.

**Comment by s-alves10** [comment]
@aimane-chnaif 

https://github.com/s-alves10/App/tree/fix/issue-17531
The branch was merged with the latest main. Please check

**Comment by techievivek** [comment]
Bumping @aimane-chnaif to test the proposal.

**Comment by techievivek** [comment]
Gentle bump @aimane-chnaif

**Comment by s-alves10** [comment]
@aimane-chnaif 
Did you test my solution?

**Comment by sonialiap** [comment]
@aimane-chnaif any update? 

I believe you've been assigned quite a few issues recently, if you don't have the bandwidth for this issue, no problem, we can reassign :)

**Comment by techievivek** [comment]
Gentle bump, @aimane-chnaif can you please prioritize reviewing the proposal today? 🙏

**Comment by techievivek** [comment]
@aimane-chnaif Going to pass this to a different C+ member so they can prioritize looking into it, as this issue has been open for quite some time now. Thanks

**Comment by techievivek** [comment]
Sorry for the ping @joekaufmanexpensify, I just re-added the `external` and `help wanted` labels so we can have a different C+ member help us review the proposal.

**Comment by techievivek** [comment]
@0xmiroslav Can you please prioritize reviewing the proposal here https://github.com/Expensify/App/issues/17531#issuecomment-1621296819

**Comment by techievivek** [comment]
Gentle bump @0xmiroslav, can you please prioritize this GH as it has been open for some time now, and we are yet to review @HezekielT's proposal.

**Comment by techievivek** [comment]
Not overdue, bumped @0xmiroslav on slack to review the proposal.https://expensify.slack.com/archives/C02NK2DQWUX/p1689225885450879

**Comment by techievivek** [comment]
@HezekielT Sorry for the delayed response, I have bumped @0xmiroslav again on Slack to prioritize review the proposal here.

**Comment by techievivek** [comment]
Adding @s77rt as the C+ here to help us move this forward 🙇

**Comment by s77rt** [comment]
@HezekielT Thanks for the proposal. This is an old issue, a lot have changed and I see the original reported bug is no longer reproducible. As far as I can tell we only have case 3 left to be fixed? I think your solution would cause a regression, consider the following scenario:
1. Open a group chat or a public room
2. Send a message
3. Start editing the message
4. Close the App
5. Come back later
6. Now the chat would already have many messages, your message won't even be visible but it's still in edit mode
7. Send another message
8. Edit that message (Main Composer is now hidden)
9. After editing that last message the Main Composer won't show because we still have another message in edit mode but we can't even see it

I think case 3 is a low value and with the current structure it's hard to get right. We have this issue https://github.com/Expensify/App/issues/15959 that should cover that but it's not something that will get fixed any soon due to some RN limitations. We can probably close this issue if case 3 is the only remaining case.

**Comment by s77rt** [comment]
@HezekielT Please do not post PRs, code diffs, etc. Can you please explain how we would know whether a message in edit mode has focus?

**Comment by s77rt** [comment]
@HezekielT
1. Switch to edit mode in message A
2. Click the cancel button of message A
3. Main Composer won't show because `shouldShowMainComposerWhenBlurred` would still be `false` as the `onBlur` won't fire in this case.

**Comment by s77rt** [comment]
@HezekielT That contradicts with the solution, if we will always show the Main Composer in `deleteDraft` then it will always be shown even if we have other messages in edit mode.

1. Switch to edit mode in message A
2. Switch to edit mode in message B
3. Focus is on message B
4. Click the cancel button of message A
5. Focus in still on message B but Main Composer is now visible

**Comment by s-alves10** [comment]
@s77rt 
Did you check my [proposal](https://github.com/Expensify/App/issues/17531#issuecomment-1589520426)?

**Comment by s77rt** [comment]
@s-alves10 Sorry, I didn't know there is other proposals to be reviewed. But I just reviewed your proposal. It's not clear when the the main composer will be shown and when it will be hidden, can you please clarify that?

> They are set to true when the corresponding message is editing mode and has focus, and set to false in other cases.

Setting to true or false is explicit - must be done on request on. You need to specify when exactly we set to true and to false. Also I think we will have a problem:
1. Switch to edit mode in message A
2. Close the App

Now message A is still considered as a message in edit mode which I think will conflict with your logic and causes this issue https://github.com/Expensify/App/issues/17531#issuecomment-1643638883.

**Comment by s-alves10** [comment]
@s77rt 

I write my idea here.
We store object value to `SHOULD_SHOW_COMPOSE_INPUT` key in onyx store. The object is the form of 
```
    {actionID: boolean}
```

We set this value to `false` when the `TextInput` of the composer get focused, and set to true when `TextInput` get blurred.
And the main composer will be shown only when there is no messages in edit mode with focus. Although several messages are in edit mode, if none of them is focused, the composer will be shown

I think this is the right logic. Will you point out the drawbacks?

**Comment by s-alves10** [comment]
No worries about https://github.com/Expensify/App/issues/17531#issuecomment-1643638883. Even though there are some messages in edit mode, we show the main composer if none of them is focused

**Comment by s77rt** [comment]
@HezekielT I think that would work, but won't that create an inconsistency? On web clicking the delete button blurs the inputs unlike on native. Besides how will we get the number of drafts? We can't subscribe `withReportActionsDrafts` for every report action, that's an overkill.

**Comment by s77rt** [comment]
@s-alves10
> Even though there are some messages in edit mode, we show the main composer if none of them is focused

What if a message is not focused but it appears as focused? As mentioned in this case https://github.com/Expensify/App/issues/17531#issuecomment-1644087124

**Comment by s-alves10** [comment]
@s77rt 
AFAIK, when the user goes to the report, last action in edit mode will be focused. Am I right?

**Comment by s77rt** [comment]
@s-alves10 I think so but only if it's within the viewport.

**Comment by s-alves10** [comment]
@s77rt 

> What if a message is not focused but it appears as focused?

This is impossible because we focus the last editing message.

On small screen devices, we set focus to the last editing message and auto scroll to it. It may take time. I think this isn't a bug, but an UX issue.

I think we need to set expectation here in such edge-cases

**Comment by s77rt** [comment]
@s-alves10 But we store the focus state in Onyx, don't we?

**Comment by s-alves10** [comment]
> @s-alves10 But we store the focus state in Onyx, don't we?

yes

**Comment by s77rt** [comment]
@HezekielT In `ReportActionItem`  we only get one draft 1 to 1. Your solution would require each report action linked to all the drafts 1 to n.

**Comment by s77rt** [comment]
@s-alves10 So we will have stale data.
1. Switch to edit mode in message A (onyx now stores this message as focused)
2. Close App (onyx will still see this message as focused)
3. Come back later
4. Open chat
5. Our message is no longer in view but per onyx it's still focused

**Comment by s-alves10** [comment]
@s77rt 

Will you let me know your expectation in this case?

**Comment by s77rt** [comment]
@s-alves10 I would expect the data to reflect the actual state. Stale data may result in unexpected behaviour as per your solution we use this data to determine whether we should show the Main Composer.

**Comment by s-alves10** [comment]
@s77rt 

> I would expect the data to reflect the actual state.

Yeah. I manage the actual state. What we need is how to determine `stale` data, I think

**Comment by s77rt** [comment]
Ultimately I would really prefer not to involve Onyx here. Are we still talking about case 3 that @HezekielT mentioned? Case 2 does not seem reproducible - there is no back button in iOS?

**Comment by s-alves10** [comment]
@s77rt 

> Ultimately I would really prefer not to involve Onyx here.

You don't want to save draft status? That is a big change and isn't the expectation of this issue. I'm not sure what you mean

**Comment by s77rt** [comment]
@s-alves10 What I meant is the solution of this issue should not relay on onyx data. It would be better to fix this in a more natural way.

**Comment by s-alves10** [comment]
@s77rt 

Please be more clear. How can we store draft status of the report without onyx store?

**Comment by s77rt** [comment]
@s-alves10 This has nothing to do with the draft status / values. They should be kept as they are. I'm referring to the fact that the Main Composer shows unexpectedly. This should be fixed without relaying on Onyx.

**Comment by s-alves10** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/17531#issuecomment-1589520426)

@s77rt 
Please check

**Comment by s77rt** [comment]
@HezekielT Thanks for the update. As I understood your solution is basically:
1. In `deleteDraft` only call `ComposerActions.setShouldShowComposeInput(true)` if we are on a web platform

This will fix the issue in native but will keep web as is (issue would still be reproducible in web)

**Comment by s77rt** [comment]
@s-alves10 Thanks for the update. The solution makes sense but it only works on native - probably due to the difference in behaviour of focus. On web, clicking on the delete button will make the input lose focus already and the main composer will never show up. Maybe we can fix that if we `preventDefault` on clicking the delete button but not sure if that's the way to go.

**Comment by s-alves10** [comment]
@s77rt 

Please check https://github.com/s-alves10/App/tree/fix/17531-new
Some updates are there for web platforms

**Comment by s77rt** [comment]
@s-alves10 @HezekielT Please do not post PRs, branches, code diffs, etc.

**Comment by s-alves10** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/17531#issuecomment-1589520426)

@s77rt 
I added some updates for web

**Comment by s77rt** [comment]
@s-alves10 Thanks for the update. Unfortunately that web additional will result in falsy data. The input won't be focused but the state will says it's still focused. I'm pretty sure we moved that line for a case like that. Can we instead prevent the input from losing focus?

**Comment by s-alves10** [comment]
Nice idea. We can do that by simply adding onMouseDown preventing default action. I'm updating the proposal

**Comment by s-alves10** [comment]
@s77rt 

Can you check the [updated proposal](https://github.com/Expensify/App/issues/17531#issuecomment-1589520426) again?

**Comment by s77rt** [comment]
@HezekielT

> @s77rt in web, clicking on save or cancel causes the edit input to be blurred. so shouldn't we always show the main composer in this case given that all edit input don't have focus?

I would say yes, but I think we can fix this inconsistency, on native clicking save or cancel keep the input focus, we can achieve the same for web.


> Also could you please tell me how it will be reproducible? I don't think it willl. Should I summurize my solution to clear any confusion you may have?

It's already reproducible in `main` and your proposal does not change the behaviour in web.

**Comment by s77rt** [comment]
@s-alves10 I think we may have a regression, if you long press on the cancel button the input will lose focus and the next click will not show the Main Composer. I think this is due to some overlapping logic https://github.com/Expensify/App/blob/2c5dd690028c11edb5bbaca134f7db663835e4a4/src/pages/home/report/ReportActionItemMessageEdit.js#L331-L334

We can probably remove this early return as it won't be needed if we prevent input from losing focus.


Also, how would you propose to fix the [second case](https://github.com/Expensify/App/issues/17531#issuecomment-1643655143)?

**Comment by s77rt** [comment]
@HezekielT The main objective is to fix the issue. But consistency matters, if we can fix the issue in all platforms we should do so. I think your updated solution is pretty much same as @s-alves10.

**Comment by s-alves10** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/17531#issuecomment-1589520426)

**Comment by s77rt** [comment]
@HezekielT I understand your concern. The proposal selection won't be based on any of my suggestions, those were posted just to see how far we can go with a given solution.

- @HezekielT's proposal: On clicking the cancel button, show the main composer only if we are on mWeb
- @s-alves10's proposal: On clicking the cancel button, show the main composer only if the current input is focused


The proposals are a bit different but achieve same result and have same problem. You updated your proposal to:
- @HezekielT's updated proposal: On clicking the cancel button, show the main composer only if we are on mWeb and the current input is focused

This is now becoming the same as @s-alves10's proposal, in fact we can drop the mWeb part now.


Please keep in mind we are still evaluating solutions so feel free to propose any solution you see fit. We are also still looking for a solution for case 2.

**Comment by s77rt** [comment]
@s-alves10 Thanks for the update. Regarding case 2, can you please explain the root cause and why it happens only when we start editing the 2nd message. Also please try to propose a solution in plain English.

**Comment by s-alves10** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/17531#issuecomment-1589520426)

**Comment by s77rt** [comment]
@s-alves10 Thanks for the update. The RCA of the second case makes sense and solution looks good to me (same thing goes with the other case). I think we can proceed with that.

:ribbon: :eyes: :ribbon: C+ reviewed
[Link to proposal](https://github.com/Expensify/App/issues/17531#issuecomment-1589520426)

___

Recap: we have 3 issues / 3 cases
## Case 1
Seems fixed already

## Case 2
- Switch to edit mode in message A
- Switch to edit mode in message B
- Hide keyboard (press back button)
- Main Composer is shown but it shouldn't because focus is still on message B

### Root cause
When we switch to edit mode in message B, message A is blurred and we register a keyboard event listener that shows the Main Composer on keyboard hide.

### Solution
When we switch to edit mode in message B, message A is blurred but message B is focused. Remove any previous keyboard event listeners on that focus event.

## Case 3
- Switch to edit mode in message A
- Switch to edit mode in message B
- Click the cancel button of message A
- Main Composer is shown but it shouldn't because focus is still on message B

### Root cause
When we click the cancel button we show the Main Composer no questions askted

### Solution
When we click the cancel button we show the Main Composer only if the current input is focused

**Comment by s77rt** [comment]
Not overdue. @techievivek Are we good moving with @s-alves10's [proposal](https://github.com/Expensify/App/issues/17531#issuecomment-1589520426)?

**Comment by techievivek** [comment]
I feel like both of you have shown a great commitment in fixing this issue though the proposal reviews were delayed here.

I've been thinking about the best way to acknowledge your efforts, and I came up with an idea. Since both of you have actually put in a great amount of effort, I think it would be really fair to share the bounty between you.

It's a win-win situation for everyone involved. I'd love to hear your thoughts on this @s77rt @HezekielT @s-alves10. Thanks 🙇

**Comment by s77rt** [comment]
I agree. Both @HezekielT and @s-alves10 worked on this for a long time and both deserve a bounty here. Although I think @s-alves10 should raise the PR to have the opportunity of the 50% bonus since we are going with his proposal.

**Comment by s-alves10** [comment]
@s77rt @techievivek 

I agree

**Comment by techievivek** [comment]
@sonialiap We have [decided](https://expensify.slack.com/archives/C056AV5V8UX/p1690223119507749?thread_ts=1688579150.685009&cid=C056AV5V8UX) to share the bounty equal between @HezekielT and @s-alves10, and @s-alves10 alone would be eligible for the 50% bonus if the PR gets merged in 3 business days.

**Comment by s-alves10** [comment]
@s77rt 

[PR](https://github.com/Expensify/App/pull/23618) is ready for review

**Comment by s77rt** [comment]
Checklist is not due yet. We reverted the PR.

@techievivek Can we open this for proposals?

**Comment by techievivek** [comment]
I think it would mess up the automation here 🤣, what would happen to the existing payment? Also, do we need to update the bounty to 1000 again?

**Comment by s77rt** [comment]
There is no payment due here. If it may be easier we can close this issue and move it to the new one?

**Comment by techievivek** [comment]
Ok, just to confirm, we don't pay anything if the PR gets reverted? I agree that moving the discussion to the new issue would be better.

**Comment by s-alves10** [comment]
There is one issue related to main composer
https://github.com/Expensify/App/issues/23908

I think it'd be good to close this issue as well and create a new issue containing all cases we've found.
I can create a bug report for this if you don't mind

**Comment by techievivek** [comment]
To summarize what happened over here, we reverted the [PR](https://github.com/Expensify/App/pull/23618) selected for this issue and we will need to investigate further to find a proper fix for the bug(it's a complex one). Therefore, we have decided to close this issue and continue the discussion or propose new solutions at https://github.com/Expensify/App/issues/23908. Additionally, I agree that a starting price of $2000 is reasonable for this bug. Thank you all for your efforts.

We have to make no payment over here.
